File: The Braithwaite Confront/sol.js
function max(a, b) {
  return a >= b ? a : b;
}

function solve(input) {
  const [n, m, r, c] = input.split(" ").map(Number);
  console.log(max(n - r, r - 1) + max(m - c, c - 1));
}

let rl = require("readline").createInterface({
  input: process.stdin,
});

rl.on("line", (input) => {
  solve(input);
  rl.close();
});

File: The Braithwaite Confront/sol.py
def max(a, b):
    return a if a >= b else b

def solve():
    n, m, r, c = map(int, input().split())
    print(max(n - r, r - 1) + max(m - c, c - 1))

if __name__ == "__main__":
    solve()
